### PR TITLE
images: Create script for /var/lib/package-sets/

### PR DESCRIPTION
**Pull request by martinpitt**
Statically downloaded binaries break our "updates-testing" scenario when any of these packages have pending updates. So create a script which creates the package sets, so that we can re-run it with updated repositories at image-prepare time.

 * [x] image-refresh fedora-39

### CODE REVIEW

**Comment by martinpitt** [comment]
This blocks https://github.com/cockpit-project/cockpit/issues/20184

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-fedora-39-7eb5c5d9-20240320-050537/log.html

**Comment by cockpituous** [comment]
image-refresh fedora-39 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-39-20240320-054321

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-fedora-39-1d7790d6-20240320-052133/log.html

**Comment by martinpitt** [comment]
We have to ignore the composer failures, they've been broken for several weeks now. @jkozol , periodic reminder, can you please look at that testCreateAndDeleteBlueprint() failure?

File: images/scripts/fedora.setup [hunk 1]
```diff
@@ -175,11 +175,19 @@ fi
 dnf $DNF_OPTS -y upgrade
 dnf $DNF_OPTS -y install $ALL_DEPS
 
+# create/run this as a script, so that it can be re-run at image-prepare time with different repos
+cat > /var/lib/download-package-sets.sh <<EOF
+#!/bin/sh
+set -eu
+rm -rf /var/lib/download-package-sets
```
Comment (mvollmer) [inline images/scripts/fedora.setup:182]: This should be `rm -rf /var/lib/package-sets`, no?

File: images/scripts/fedora.setup [hunk 2]
```diff
@@ -175,11 +175,19 @@ fi
...
 mkdir -p /var/lib/package-sets/clevis
-dnf $DNF_OPTS download --destdir=/var/lib/package-sets/clevis $PACKAGE_SET_CLEVIS
+dnf \${DNF_OPTS:=} download --destdir=/var/lib/package-sets/clevis $(echo "$PACKAGE_SET_CLEVIS" | xargs)
```
Comment (mvollmer) [inline images/scripts/fedora.setup:184]: Hmm, what does the escaping do here? Don't we want to insert the options into the script? `DNF_OPTS` will not be defined when the script is run, no?

File: images/scripts/fedora.setup [hunk 3]
```diff
@@ -175,11 +175,19 @@ fi
...
 
 mkdir -p /var/lib/package-sets/stratis
-dnf $DNF_OPTS download --destdir=/var/lib/package-sets/stratis $PACKAGE_SET_STRATIS
+dnf \$DNF_OPTS download --destdir=/var/lib/package-sets/stratis $(echo "$PACKAGE_SET_STRATIS" | xargs)
```
Comment (mvollmer) [inline images/scripts/fedora.setup:187]: This is different from the `\${DNF_OPTS:=}` above. Is there a reason?

**Review by mvollmer — CHANGES_REQUESTED** [review]

File: images/scripts/fedora.setup (hunk 1 above)
Comment (martinpitt) [reply images/scripts/fedora.setup]: Right, thanks! Fixed

File: images/scripts/fedora.setup (hunk 2 above)
Comment (martinpitt) [reply images/scripts/fedora.setup]: I misunderstood this. I thought it was a knob for using this from "outside", but indeed once upon a time fedora.setup actually defined `$DNF_OPTS`. It doesn't any more, and the script is not `set -u`. I fixed that, too.

File: images/scripts/fedora.setup (hunk 3 above)
Comment (martinpitt) [reply images/scripts/fedora.setup]: There was, `:=` sets a default value if the variable is unset. But this is moot now, see the previous commit.

**Comment by cockpituous** [comment]
image-refresh fedora-39 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-39-20240320-072337

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-fedora-39-40a2f0b9-20240320-065523/log.html

**Review by mvollmer — APPROVED** [review]
Danke!

**Pull request merged**